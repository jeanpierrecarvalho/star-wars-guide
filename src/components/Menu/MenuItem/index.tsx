import Image from "next/image";
import Link from "next/link";

interface IProps {
  href: string;
  imageSrc: string;
  imageAlt: string;
  title: string;
}

const MenuItem: React.FC<IProps> = ({ href, imageSrc, imageAlt, title }) => {
  return (
    <div
      className="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <Link href={href} passHref aria-label={`Link to ${title}`}>
        <div className="cursor-pointer">
          <div className="flex flex-row justify-center">
            <Image
              src={imageSrc}
              width={300}
              height={300}
              alt={imageAlt}
              priority={true}
            />
          </div>
          <div className="flex flex-row justify-center mt-4">
            <span
              className="font-starjout text-4xl text-center text-yellow-500"
              aria-label={`Title: ${title}`}
            >
              {title}
            </span>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default MenuItem;
