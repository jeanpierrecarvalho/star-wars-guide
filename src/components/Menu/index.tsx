import React, { memo } from "react";
import { ENTITY_TYPE } from "@/constants/entities";
import MenuItem from "@/components/Menu/MenuItem";

const Menu: React.FC = () => {
  return (
    <div
      className="justify-center grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 mt-8"
      role="navigation"
      aria-label="Main Menu"
    >
      {Object.entries(ENTITY_TYPE).map(([key, { entity, img }]) => (
        <MenuItem
          key={key}
          href={`/${entity}`}
          imageSrc={img.src}
          imageAlt={img.alt}
          title={entity}
        />
      ))}
    </div>
  );
};

export default memo(Menu);
