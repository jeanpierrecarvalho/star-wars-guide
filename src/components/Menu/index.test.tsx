import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";

import Menu from "@/components/Menu";

describe("Menu Component", () => {
  it("matches the snapshot", () => {
    const { container } = render(<Menu />);
    expect(container).toMatchSnapshot();
  });
});
