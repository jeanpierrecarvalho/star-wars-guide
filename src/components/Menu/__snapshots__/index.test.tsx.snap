// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu Component > matches the snapshot 1`] = `
<div>
  <nav
    aria-label="Main Menu"
    class="justify-center grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 mt-8"
    role="menu"
  >
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to characters"
        href="/characters"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Characters Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Fcharacters.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Fcharacters.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fcharacters.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: characters"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              characters
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to planets"
        href="/planets"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Planets Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Fplanets.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Fplanets.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fplanets.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: planets"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              planets
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to films"
        href="/films"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Films Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Ffilms.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Ffilms.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Ffilms.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: films"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              films
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to species"
        href="/species"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Species Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Fspecies.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Fspecies.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fspecies.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: species"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              species
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to starships"
        href="/starships"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Starships Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Fstarships.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Fstarships.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fstarships.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: starships"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              starships
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      class="mb-12 hover:scale-105 opacity-90 hover:opacity-100 py-4"
      role="menuitem"
    >
      <a
        aria-label="Link to vehicles"
        href="/vehicles"
      >
        <div
          class="cursor-pointer"
        >
          <div
            class="flex flex-row justify-center"
          >
            <img
              alt="Vehicles Image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Fimages%2Fvehicles.png&w=640&q=75"
              srcset="/_next/image?url=%2Fimages%2Fvehicles.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fvehicles.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
          <div
            class="flex flex-row justify-center mt-4"
          >
            <span
              aria-label="Title: vehicles"
              class="font-starjout text-4xl text-center text-yellow-500"
            >
              vehicles
            </span>
          </div>
        </div>
      </a>
    </div>
  </nav>
</div>
`;
